File: ObjectiveC/FlexChart101/FlexChart101/FinancialData.m
//
//  FinancialData.m
//  FlexChart101
//
//  Copyright (c) 2015 GrapeCity. All rights reserved.
//

#import "FinancialData.h"

@implementation FinancialData
- (id)initWithDate: (NSDate *)date open: (NSNumber *)open high: (NSNumber *)high low: (NSNumber *)low close: (NSNumber *)close volume: (NSNumber *)volume {
    self = [super init];
    if(self){
        _date = date;
        _open = open;
        _high = high;
        _low = low;
        _close = close;
        _volume = volume;
    }
    return self;
}
+ (NSNumber *) generateRandom: (NSInteger) max {
    return [NSNumber numberWithUnsignedInteger:(arc4random() % max)];
}

+ (NSMutableArray *) demoData {
    NSDate *date = [NSDate date];
    NSDate *previousDate;
    NSNumber *open = [[NSNumber alloc] init];
    NSNumber *high = [[NSNumber alloc] init];
    NSNumber *low = [[NSNumber alloc] init];
    NSNumber *close = [[NSNumber alloc] init];
    NSNumber *volume = [[NSNumber alloc] init];
    NSDateComponents *dateComps = [[NSDateComponents alloc] init];
    NSCalendar *calendar = [NSCalendar currentCalendar];
    NSMutableArray *array = [[NSMutableArray alloc] init];
    FinancialData *yesterday;
    for (int i = 0; i < 25; i++) {
        if(i > 0){
            yesterday = (FinancialData*)[array objectAtIndex:(i - 1)];
            open = yesterday.close;
        }
        else{
            open = @1000.0;
        }
        dateComps.day = i;
        previousDate = [calendar dateByAddingComponents:dateComps toDate:date options:false];
        high = @([open integerValue] + [[FinancialData generateRandom:21] integerValue]);
        low = @([open integerValue] - [[FinancialData generateRandom:21] integerValue]);
        
        NSInteger temp = [high integerValue] == [low integerValue] ? 1 : [high integerValue] - [low integerValue];
        close = @([low integerValue] + [[FinancialData generateRandom:temp] integerValue]);
        volume = [FinancialData generateRandom:101];
        FinancialData *finance = [[FinancialData alloc]initWithDate:previousDate open:open high:high low:low close:close volume:volume];
        [array addObject:finance];
    }
    return array;
}
@end

File: Swift/FlexChart101/FlexChart101/FinancialData.swift
//
//  FinancialData.swift
//  FlexChart101
//
//  Copyright (c) 2015 GrapeCity. All rights reserved.
//

import UIKit

class FinancialData: NSObject {
    var date: NSDate
    var open: Double
    var high: Double
    var low: Double
    var close: Double
    var volume: Double
    
    init(date: NSDate, open: Double, high: Double, low: Double, close: Double, volume: Double) {
        self.date = date
        self.open = open
        self.high = high
        self.low = low
        self.close = close
        self.volume = volume
    }
    
    class func demoData() -> NSMutableArray {
        var array = NSMutableArray()
        var previousDate = NSDate()
        var now = NSDate()
        var open, high, low, close, volume : Double
        var yesterday : FinancialData
        
        for var i = 0; i < 25; i++ {
            if(i > 0){
                yesterday = array.objectAtIndex(i - 1) as! FinancialData
                open = yesterday.close
            }
            else{
                open = 1000
            }
            
            previousDate = now.dateByAddingTimeInterval(24 * 60 * 60 * Double(i))
            high = open + Double(arc4random() % 21)
            low = open - Double(arc4random() % 21)
            close = low + Double(arc4random() % UInt32(high == low ? 1 : high - low))
            volume = Double(arc4random() % 101)
            let finance = FinancialData(date: previousDate, open: open, high: high, low: low, close: close, volume: volume)
            array.addObject(finance)
        }
        
        return array
    }
}
